src/methods.ts

```diff
@@ -254,6 +347,11 @@ export class Method extends AbstractMethodField {
    */
   private code: any;
 
+  private numOpcodeExecs = 0;
+  private jitThreshold = 0;
+  private compiledFunctions: Function[] = [];
```

Comment:
If you made this a map from `startopcode` to `string`s with the implementation of that basic block, your JIT could produce a single `run` function that does something like:

```{js}
switch (pc) {
  case 10:
    ${this.compiledBlocks[10]}
    break;
  case 20:
    ${this.compiledBlocks[20]}
    break;
  default:
    this.interpret();
}
```